.github/workflows/typescript_and_e2e_coverage.yml

```diff
@@ -24,28 +24,23 @@ jobs:
           python-version: '3.8.15'
           architecture: 'x64'
       - uses: ./.github/actions/merge
-      - name: Cache node modules and third_party/static
-        uses: actions/cache@v3
-        env:
-          cache-name: cache-node-modules
-        with:
-          path: |
-            /home/runner/work/oppia/yarn_cache
-            /home/runner/work/oppia/oppia/third_party/static
-          key: ${{ runner.os }}-build-${{ env.cache-name }}-${{ hashFiles('yarn.lock', 'dependencies.json') }}
-          restore-keys: |
-            ${{ runner.os }}-build-${{ env.cache-name }}-
-            ${{ runner.os }}-build-
-            ${{ runner.os }}-
-      - uses: ./.github/actions/install-oppia-dependencies
+      # Installing python dependencies becuase we are executing a python script for installing chrome in CI.
+      - name: Install Python deps
+        run: |
+          pip install --require-hashes --no-deps -r requirements.txt
+          pip install --require-hashes --no-deps -r requirements_dev.txt
       - name: Install chrome
         run: python -m scripts.install_chrome_for_ci
+      - name: Initialize Containers
+        run: |
+          sudo make build
```

Comment:
this will build the frontend and backend dockerfiles only. we need this build step since we need to seperate the ddocker logs from the test logs, so using seperate step

Reply:
Why do we need to build the backend image? 

> we need this build step since we need to seperate the ddocker logs from the test logs, so using seperate step

That's nice, but we should not build unnecesary images.

Reply:
backend image is used for the dev-server container and every test is running in this container.
frontend image is installing node_modules, we need to have node_modules present for the typescript tests